theano/sandbox/gpuarray/basic_ops.py

```diff
@@ -343,43 +382,93 @@ def c_code(self, node, name, inputs, outputs, sub):
                                      PyArray_NDIM(%(inp)s),
                                      (size_t *)PyArray_DIMS(%(inp)s),
                                      (ssize_t *)PyArray_STRIDES(%(inp)s),
-                                     pygpu_default_context(),
+                                     %(ctx)s,
                                      Py_None);
         if (%(out)s == NULL) {
             %(fail)s
         }
-        """ % {'name': name, 'inp': inputs[0],
+        """ % {'name': name, 'inp': inputs[0], 'ctx': sub['context'],
                'out': outputs[0], 'fail': sub['fail']}
 
     def c_code_cache_version(self):
-        return (4,)
+        return (4, 1)
```

Comment:
Because this allows experimentation before upgrading to 5 for the final version.  I should fix that and all similar version.  I'll do that at the same time as the terminology change.